lib/hashicorptools/code_deploy.rb

```diff
@@ -7,16 +7,59 @@
 require 'thor'
 
 module Hashicorptools
+  class RegionDeployment
+    attr_accessor :aws_region, :environment
+
+    def initialize(aws_region:, environment:)
+      @aws_region = aws_region
+      @environment = environment
+    end
+
+    def create_deployment(commit_id, commit_message)
+      Dotenv.load
+
+      client = Aws::CodeDeploy::Client.new(region: aws_region)
+      response = client.create_deployment({
+                                            application_name: application_name,
+                                            deployment_group_name: "#{application_name}-#{@environment}",
+                                            revision: {
+                                              revision_type: 'GitHub',
+                                              git_hub_location: {
+                                                repository: "controlshift/#{application_name}",
+                                                commit_id: commit_id
+                                              }
+                                            },
+                                            description: (commit_message || "commit #{commit_id}").slice(0,99)
+                                          })
+      output "created deployment #{response.deployment_id}"
+      output "https://console.aws.amazon.com/codedeploy/home?region=#{aws_region}#/deployments/#{response.deployment_id}"
+    end
+
+    private
+
+    def application_name
+      raise "implement me"
+    end
+
+    def output(text)
+      puts "[#{aws_region}] #{text}"
+    end
+  end
+
   class CodeDeploy < Thor
+    AWS_REGION_US_EAST_1 = 'us-east-1'
 
     desc 'deploy', 'deploy latest code to environment'
     option :environment, :required => true
     option :branch
-    option :aws_region, default: 'us-east-1'
+    option :aws_regions, :type => :array
```

Comment:
I think newer colon syntax for hashes with symbol keys is still preferred for symbol values

```suggestion
    option :aws_regions, type: :array
```